src/tiled/gotodialog.cpp

```diff
@@ -0,0 +1,94 @@
+#include "gotodialog.h"
+
+#include <QApplication>
+#include <QHBoxLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QPushButton>
+#include <QIntValidator>
+#include <QString>
+#include "documentmanager.h"
+#include "mapview.h"
+#include "mapscene.h"
+#include "mapreader.h"
+#include "mapdocument.h"
+#include "maprenderer.h"
+
+using namespace Tiled::Internal;
+using namespace Tiled;
+
+GotoDialog *GotoDialog::mInstance;
+
+GotoDialog::GotoDialog(QWidget *parent, Qt::WindowFlags f )
+    : QDialog(parent, f)
+{
+    setWindowTitle(tr("Go to"));
+
+    QGroupBox* horizontalGroupBox = new QGroupBox(tr("Go to tile:"));
+    QHBoxLayout *layout = new QHBoxLayout;
+    horizontalGroupBox->setLayout(layout);
+    QLabel* labelX = new QLabel(tr("X:"));
+    QLabel* labelY = new QLabel(tr("Y:"));
+    lineEditX = new QLineEdit(tr("0"));
+    lineEditY = new QLineEdit(tr("0"));
+    QPushButton* goButton = new QPushButton(tr("Go"));
+
+    lineEditX->setMaximumWidth(50);
+    lineEditY->setMaximumWidth(50);
+    lineEditX->setValidator(new QIntValidator);
+    lineEditY->setValidator(new QIntValidator);
+
+    layout->addWidget(labelX);
+    layout->addWidget(lineEditX);
+    layout->addWidget(labelY);
+    layout->addWidget(lineEditY);
+    layout->addWidget(goButton);
+
+    QVBoxLayout *mainLayout = new QVBoxLayout;
+    mainLayout->addWidget(horizontalGroupBox);
+
+    connect(goButton, &QAbstractButton::clicked, this, &GotoDialog::goToCoordinates);
+
+    setLayout(mainLayout);
+}
+
+GotoDialog* GotoDialog::showDialog()
+{
+    if (!mInstance) {
+        QWidget* parentWidget = QApplication::activeWindow();
+        mInstance = new GotoDialog(parentWidget, Qt::Tool);
+    }
+
+    mInstance->show();
+    mInstance->raise();
+    mInstance->activateWindow();
+
+    return mInstance;
+}
+
+void GotoDialog::goToCoordinates()
+{
+    QString textX = lineEditX->text();
+    QString textY = lineEditY->text();
+
+    MapView *mapView = DocumentManager::instance()->currentMapView();
+
+    if (mapView) {
+        MapDocument *mapDocument = mapView->mapScene()->mapDocument();
+        MapRenderer *renderer = mapDocument->renderer();
+        QPointF point = renderer->tileToScreenCoords(textX.toDouble(),textY.toDouble());
+        QSize size = mapView->viewport()->geometry().size();
+        QPoint origin(point.x() - size.width()/2.0, point.y() - size.height()/2.0f);
+
+        QRectF rect(origin,size);
+
+        if (!mapView->sceneRect().contains(rect)) {
+            QRectF newRect = mapView->sceneRect().united(rect);
+            mapView->setSceneRect(newRect);
```

Comment:
The issue with changing the scene rect is that it is permanent (until the scene rect gets refreshed, for example when switching maps). Whereas when scrolling past the boundaries, the scroll bar range gets temporarily extended and shrinks again as you scroll back to the map.

> I did try only by using forceCenterOn, the only problem is that if the scene rect is not big enough it would only make an approximation, so if the tile was in the upper left cornern it would scroll to the limit of the view without centering the tile.

There might be an issue with `forceCenterOn`, though it seems to be working fine for the minimap.

> Beside this a nice addition would be to add a temporaly highlight to the tile since even with the tile exactly in the center it's hard to identify it.

Nice idea! Feel free to open an issue about it or make it your next pull request. :-)